EXPLORING CRIME RATES BY RACE IN THE UNITED STATES

INTRODUCTION

The incarcerated population of the US has seen a dramatic increase from the mid-1970s (Weidner & Schultz, 2019). However, there is a disparity between incarceration rates by race; for example, Black and Hispanic individuals are incarcerated at six and three times the rate of Whites, respectively, in state and federal prisons. As a result, we are interested in investigating the relationship between different communities and their crime rates by violent and non-violent types. From that, we can look closer into the demographic of each community and further examine the relationships between crime and race. The big question to be answered is therefore, “Can the ratio of violent crime rate to non-violent crime rate in a community be predicted based on the racial characteristics of the community?”.

In order to answer this question, the "Communities and Crimes" dataset from the UCI Machine Learning Repository was chosen, which contains various socio-economic, law enforcement and crime statistics from the 1990 US Census, 1990 US LEMAS survey and 1995 FBI UCR (Dua & Graff, 2019). The data is unnormalized and observations are listed at the community level, which includes cities, townships, and boroughs. The scope of the variables also needs to be addressed, due to the multitude of crime types that exist. For this dataset, violent crimes consist of murder, rape, robbery, and assault. Non-violent crimes include burglary, larceny, auto theft and arson.

METHODOLOGY

We aim to predict the ratio of violent to non-violent crime rate by race, hence our target and predictor variables will be the former and the latter, respectively. Due to the numerical variables, regression is appropriate for this question. A scatter plot will be used to visualize the relationship between the two chosen variables. The predictor variable will be placed on the x-axis and the target variable will be placed on the y-axis. Regression can then be performed to predict the y value in relation to a given value of x. 

PRELIMINARY EXPLORATORY DATA ANALYSIS

We begin by loading in the necessary packages for data wrangling and visualization, and setting the seed for our analysis.

In [1]:
# Loads packages
library(tidyverse)
library(repr)
library(tidymodels)

# Set seed
set.seed(100)

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

── Attaching packages ────────────────────────────────────── tidymodels 1.0.0 ──

✔ broom        1.0.0     ✔ rsample      1.0.0
✔ dials        1.0.0     ✔ tune         1.0.0
✔ infer        1.0.2     ✔ workflows    1.0.0
✔ modeldata    1.0.0     ✔ workflowsets 1.0.0
✔ parsnip      1.0.0     ✔ yardstick    1.0.0
✔ recipes      1.0.1     

── Conflicts ───────────────────────────────────────── tidymodels_conflicts() ──
✖ scales::discard() masks purrr::discard()
✖ dplyr::filter()   masks stats::filter()
✖ recipes::fixed()  masks stringr::fixed()
✖ dplyr::lag()      masks stats::lag()
✖ yardstick::spec() masks r

Next, we need to read in the data, using the column names from the attribute information for the dataset (Dua & Graff, 2019).

In [2]:
# Vector for column names
colnames <- c("communityname",
                "state",
                "countyCode",
                "communityCode",
                "fold",
                "population",
                "householdsize",
                "racepctblack",
                "racePctWhite",
                "racePctAsian",
                "racePctHisp",
                "agePct12t21",
                "agePct12t29",
                "agePct16t24",
                "agePct65up",
                "numbUrban",
                "pctUrban",
                "medIncome",
                "pctWWage",
                "pctWFarmSelf",
                "pctWInvInc",
                "pctWSocSec",
                "pctWPubAsst",
                "pctWRetire",
                "medFamInc",
                "perCapInc",
                "whitePerCap",
                "blackPerCap",
                "indianPerCap",
                "AsianPerCap",
                "OtherPerCap",
                "HispPerCap",
                "NumUnderPov",
                "PctPopUnderPov",
                "PctLess9thGrade",
                "PctNotHSGrad",
                "PctBSorMore",
                "PctUnemployed",
                "PctEmploy",
                "PctEmplManu",
                "PctEmplProfServ",
                "PctOccupManu",
                "PctOccupMgmtProf",
                "MalePctDivorce",
                "MalePctNevMarr",
                "FemalePctDiv",
                "TotalPctDiv",
                "PersPerFam",
                "PctFam2Par",
                "PctKids2Par",
                "PctYoungKids2Par",
                "PctTeen2Par",
                "PctWorkMomYoungKids",
                "PctWorkMom",
                "NumKidsBornNeverMar",
                "PctKidsBornNeverMar",
                "NumImmig",
                "PctImmigRecent",
                "PctImmigRec5",
                "PctImmigRec8",
                "PctImmigRec10",
                "PctRecentImmig",
                "PctRecImmig5",
                "PctRecImmig8",
                "PctRecImmig10",
                "PctSpeakEnglOnly",
                "PctNotSpeakEnglWell",
                "PctLargHouseFam",
                "PctLargHouseOccup",
                "PersPerOccupHous",
                "PersPerOwnOccHous",
                "PersPerRentOccHous",
                "PctPersOwnOccup",
                "PctPersDenseHous",
                "PctHousLess3BR",
                "MedNumBR",
                "HousVacant",
                "PctHousOccup",
                "PctHousOwnOcc",
                "PctVacantBoarded",
                "PctVacMore6Mos",
                "MedYrHousBuilt",
                "PctHousNoPhone",
                "PctWOFullPlumb",
                "OwnOccLowQuart",
                "OwnOccMedVal",
                "OwnOccHiQuart",
                "OwnOccQrange",
                "RentLowQ",
                "RentMedian",
                "RentHighQ",
                "RentQrange",
                "MedRent",
                "MedRentPctHousInc",
                "MedOwnCostPctInc",
                "MedOwnCostPctIncNoMtg",
                "NumInShelters",
                "NumStreet",
                "PctForeignBorn",
                "PctBornSameState",
                "PctSameHouse85",
                "PctSameCity85",
                "PctSameState85",
                "LemasSwornFT",
                "LemasSwFTPerPop",
                "LemasSwFTFieldOps",
                "LemasSwFTFieldPerPop",
                "LemasTotalReq",
                "LemasTotReqPerPop",
                "PolicReqPerOffic",
                "PolicPerPop",
                "RacialMatchCommPol",
                "PctPolicWhite",
                "PctPolicBlack",
                "PctPolicHisp",
                "PctPolicAsian",
                "PctPolicMinor",
                "OfficAssgnDrugUnits",
                "NumKindsDrugsSeiz",
                "PolicAveOTWorked",
                "LandArea",
                "PopDens",
                "PctUsePubTrans",
                "PolicCars",
                "PolicOperBudg",
                "LemasPctPolicOnPatr",
                "LemasGangUnitDeploy",
                "LemasPctOfficDrugUn",
                "PolicBudgPerPop",
                "murders",
                "murdPerPop",
                "rapes",
                "rapesPerPop",
                "robberies",
                "robbbPerPop",
                "assaults",
                "assaultPerPop",
                "burglaries",
                "burglPerPop",
                "larcenies",
                "larcPerPop",
                "autoTheft",
                "autoTheftPerPop",
                "arsons",
                "arsonsPerPop",
                "ViolentCrimesPerPop",
                "nonViolPerPop")

In [3]:
# Reads data in and specifies column names
crime <- read_csv("https://archive.ics.uci.edu/ml/machine-learning-databases/00211/CommViolPredUnnormalizedData.txt",
                  col_names = colnames)
                                
crime

Warning message:
“One or more parsing issues, see `problems()` for details”
Rows: 2215 Columns: 147
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (42): communityname, state, countyCode, communityCode, LemasSwornFT, Le...
dbl (105): fold, population, householdsize, racepctblack, racePctWhite, race...

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


communityname           state countyCode communityCode fold population
1    BerkeleyHeightstownship NJ    39         5320          1      11980   
2    Marpletownship          PA    45         47616         1      23123   
3    Tigardcity              OR    ?          ?             1      29344   
4    Gloversvillecity        NY    35         29443         1      16656   
5    Bemidjicity             MN    7          5068          1      11245   
6    Springfieldcity         MO    ?          ?             1     140494   
7    Norwoodtown             MA    21         50250         1      28700   
8    Andersoncity            IN    ?          ?             1      59459   
9    Fargocity               ND    17         25700         1      74111   
10   Wacocity                TX    ?          ?             1     103590   
11   Shermancity             TX    ?          ?             1      31601   
12   SanPablocity            CA    ?          ?             1      25158   
13   BowlingGreencity        KY    ?          ?             1      40641   
14   PineBluffcity           AR    ?          ?             1      57140   
15   NewUlmcity              MN    15         46042         1      13132   
16   Maplewoodcity           MN    123        40382         1      30954   
17   Enfieldtown             CT    3          25990         1      45532   
18   Glendalecity            CA    ?          ?             1     180038   
19   Worthingtoncity         OH    ?          ?             1      14869   
20   Arlingtoncity           TX    ?          ?             1     261721   
21   Plymouthcity            MN    53         51730         1      50889   
22   NewYorkcity             NY    ?          ?             1    7322564   
23   Marinacity              CA    ?          ?             1      26436   
24   Lebanoncity             NH    9          41300         1      12183   
25   Rockledgecity           FL    ?          ?             1      16023   
26   Rogerscity              AR    ?          ?             1      24692   
27   Bellairecity            TX    ?          ?             1      13842   
28   ElCajoncity             CA    ?          ?             1      88693   
29   MosesLakecity           WA    ?          ?             1      11235   
30   WestMemphiscity         AR    ?          ?             1      28259   
⋮    ⋮                       ⋮     ⋮          ⋮             ⋮    ⋮         
2186 Jacksoncity             TN    ?          ?             10   48949     
2187 PalmDesertcity          CA    ?          ?             10   23252     
2188 Phoenixvilleborough     PA    29         60120         10   15066     
2189 Princetontownship       NJ    21         60915         10   13198     
2190 Herculescity            CA    ?          ?             10   16829     
2191 Wyckofftownship         NJ    3          83050         10   15372     
2192 LittleEggHarbortownship NJ    29         40560         10   13333     
2193 Teanecktownship         NJ    3          72360         10   37825     
2194 Edinburgcity            TX    ?          ?             10   29885     
2195 Hillsborocity           OR    ?          ?             10   37520     
2196 PortHuenemecity         CA    ?          ?             10   20319     
2197 Tuscaloosacity          AL    ?          ?             10   77759     
2198 Madisoncity             AL    ?          ?             10   14904     
2199 Hammontontown           NJ    1          29430         10   12208     
2200 WestMilfordtownship     NJ    31         79460         10   25430     
2201 FallRivercity           MA    5          23000         10   92703     
2202 EggHarbortownship       NJ    1          20290         10   24544     
2203 Brightoncity            CO    ?          ?             10   14203     
2204 Lawndalecity            CA    ?          ?             10   27331     
2205 Puyallupcity            WA    ?          ?             10   23875     
2206 Weatherfordcity         TX    ?          ?      

As our analysis will only consist of the ratio of violent crime rate to non-violent crime rate, the columns for the individual crimes can be ignored in favour of the total violent and non-violent rates per capita. We also will select the population percentages by race, as these will be our predictor variables. Additionally, for the sake of readability, the names of the communities and their states will be included.

In [4]:
crime_selected <- crime |>
                    select(communityname, state, racepctblack:racePctHisp, ViolentCrimesPerPop, nonViolPerPop)
crime_selected

communityname           state racepctblack racePctWhite racePctAsian
1    BerkeleyHeightstownship NJ     1.37        91.78         6.50       
2    Marpletownship          PA     0.80        95.57         3.44       
3    Tigardcity              OR     0.74        94.33         3.43       
4    Gloversvillecity        NY     1.70        97.35         0.50       
5    Bemidjicity             MN     0.53        89.16         1.17       
6    Springfieldcity         MO     2.51        95.65         0.90       
7    Norwoodtown             MA     1.60        96.57         1.47       
8    Andersoncity            IN    14.20        84.87         0.40       
9    Fargocity               ND     0.35        97.11         1.25       
10   Wacocity                TX    23.14        67.60         0.92       
11   Shermancity             TX    12.63        83.22         0.77       
12   SanPablocity            CA    21.34        49.42        17.21       
13   BowlingGreencity        KY    12.18        86.39         1.12       
14   PineBluffcity           AR    53.52        45.65         0.49       
15   NewUlmcity              MN     0.06        99.21         0.47       
16   Maplewoodcity           MN     2.52        94.39         2.03       
17   Enfieldtown             CT     2.65        95.72         1.04       
18   Glendalecity            CA     1.30        74.02        14.14       
19   Worthingtoncity         OH     2.28        94.74         2.67       
20   Arlingtoncity           TX     8.41        82.64         3.92       
21   Plymouthcity            MN     1.61        95.66         2.04       
22   NewYorkcity             NY    28.71        52.26         7.00       
23   Marinacity              CA    18.97        53.60        20.84       
24   Lebanoncity             NH     0.41        97.55         1.55       
25   Rockledgecity           FL    13.79        83.94         1.42       
26   Rogerscity              AR     0.06        97.72         0.77       
27   Bellairecity            TX     0.41        94.65         1.98       
28   ElCajoncity             CA     2.92        87.36         2.82       
29   MosesLakecity           WA     1.89        82.45         1.82       
30   WestMemphiscity         AR    42.15        56.94         0.52       
⋮    ⋮                       ⋮     ⋮            ⋮            ⋮           
2186 Jacksoncity             TN    40.25        59.13         0.44       
2187 PalmDesertcity          CA     0.96        91.35         1.81       
2188 Phoenixvilleborough     PA     6.50        91.93         1.13       
2189 Princetontownship       NJ     6.88        84.07         7.39       
2190 Herculescity            CA    12.78        39.81        43.31       
2191 Wyckofftownship         NJ     0.51        96.35         2.93       
2192 LittleEggHarbortownship NJ     0.68        98.21         0.52       
2193 Teanecktownship         NJ    26.16        66.61         5.62       
2194 Edinburgcity            TX     0.47        74.25         0.44       
2195 Hillsborocity           OR     0.49        88.58         2.20       
2196 PortHuenemecity         CA     5.55        73.53         6.80       
2197 Tuscaloosacity          AL    35.49        62.85         1.34       
2198 Madisoncity             AL     7.81        88.63         2.68       
2199 Hammontontown           NJ     1.14        94.86         0.70       
2200 WestMilfordtownship     NJ     1.03        97.20         0.91       
2201 FallRivercity           MA     1.03        97.17         1.33       
2202 EggHarbortownship       NJ     9.34        86.88         2.46       
2203 Brightoncity            CO     1.08        82.30         1.41       
2204 Lawndalecity            CA     8.32        60.62        12.11       
2205 Puyallupcity            WA     0.75        94.72         2.74       
2206 Weatherfordcity         TX     2.73        92.39         0.63       
2207 Duncancity              OK     4.26        89.42         0.54       
2208 West

The ratio of the violent crime rate to the non-violent crime rate can be calculated by dividing the violent crime rate by the non-violent crime rate. However, two columns are of the character type—they need to be converted to the double type as the values contain decimals. After this is performed, the ratio can be calculated.

In [5]:
crime_w_ratio <- crime_selected |>
                mutate(ViolentCrimesPerPop = as.numeric(ViolentCrimesPerPop),
                       nonViolPerPop = as.numeric(nonViolPerPop)) |>
                mutate(crime_ratio = ViolentCrimesPerPop / nonViolPerPop)

crime_w_ratio

Warning message in mask$eval_all_mutate(quo):
“NAs introduced by coercion”
Warning message in mask$eval_all_mutate(quo):
“NAs introduced by coercion”


communityname           state racepctblack racePctWhite racePctAsian
1    BerkeleyHeightstownship NJ     1.37        91.78         6.50       
2    Marpletownship          PA     0.80        95.57         3.44       
3    Tigardcity              OR     0.74        94.33         3.43       
4    Gloversvillecity        NY     1.70        97.35         0.50       
5    Bemidjicity             MN     0.53        89.16         1.17       
6    Springfieldcity         MO     2.51        95.65         0.90       
7    Norwoodtown             MA     1.60        96.57         1.47       
8    Andersoncity            IN    14.20        84.87         0.40       
9    Fargocity               ND     0.35        97.11         1.25       
10   Wacocity                TX    23.14        67.60         0.92       
11   Shermancity             TX    12.63        83.22         0.77       
12   SanPablocity            CA    21.34        49.42        17.21       
13   BowlingGreencity        KY    12.18        86.39         1.12       
14   PineBluffcity           AR    53.52        45.65         0.49       
15   NewUlmcity              MN     0.06        99.21         0.47       
16   Maplewoodcity           MN     2.52        94.39         2.03       
17   Enfieldtown             CT     2.65        95.72         1.04       
18   Glendalecity            CA     1.30        74.02        14.14       
19   Worthingtoncity         OH     2.28        94.74         2.67       
20   Arlingtoncity           TX     8.41        82.64         3.92       
21   Plymouthcity            MN     1.61        95.66         2.04       
22   NewYorkcity             NY    28.71        52.26         7.00       
23   Marinacity              CA    18.97        53.60        20.84       
24   Lebanoncity             NH     0.41        97.55         1.55       
25   Rockledgecity           FL    13.79        83.94         1.42       
26   Rogerscity              AR     0.06        97.72         0.77       
27   Bellairecity            TX     0.41        94.65         1.98       
28   ElCajoncity             CA     2.92        87.36         2.82       
29   MosesLakecity           WA     1.89        82.45         1.82       
30   WestMemphiscity         AR    42.15        56.94         0.52       
⋮    ⋮                       ⋮     ⋮            ⋮            ⋮           
2186 Jacksoncity             TN    40.25        59.13         0.44       
2187 PalmDesertcity          CA     0.96        91.35         1.81       
2188 Phoenixvilleborough     PA     6.50        91.93         1.13       
2189 Princetontownship       NJ     6.88        84.07         7.39       
2190 Herculescity            CA    12.78        39.81        43.31       
2191 Wyckofftownship         NJ     0.51        96.35         2.93       
2192 LittleEggHarbortownship NJ     0.68        98.21         0.52       
2193 Teanecktownship         NJ    26.16        66.61         5.62       
2194 Edinburgcity            TX     0.47        74.25         0.44       
2195 Hillsborocity           OR     0.49        88.58         2.20       
2196 PortHuenemecity         CA     5.55        73.53         6.80       
2197 Tuscaloosacity          AL    35.49        62.85         1.34       
2198 Madisoncity             AL     7.81        88.63         2.68       
2199 Hammontontown           NJ     1.14        94.86         0.70       
2200 WestMilfordtownship     NJ     1.03        97.20         0.91       
2201 FallRivercity           MA     1.03        97.17         1.33       
2202 EggHarbortownship       NJ     9.34        86.88         2.46       
2203 Brightoncity            CO     1.08        82.30         1.41       
2204 Lawndalecity            CA     8.32        60.62        12.11       
2205 Puyallupcity            WA     0.75        94.72         2.74       
2206 Weatherfordcity         TX     2.73        92.39         0.63       
2207 Duncancity              OK     4.26        89.42         0.54       
2208 West

To perform regression, the dataset needs to be split into training and testing datasets. We will set aside 75% of our data to be our training dataset, and specify the crime_ratio column to be our target variable.

In [6]:
crime_split <- crime_w_ratio |>
                initial_split(prop = 0.75, strata = crime_ratio)

crime_training <- training(crime_split)
crime_testing <- testing(crime_split)

To obtain a general overview of the training dataset and help build the model, summary statistics will be useful. We will exclude observations with missing values, thus, finding the number of observations with missing values will help us ensure that there is a sufficient quantity of data to train the model. It would also be helpful to find the mean and range of the racial percentages and crime_ratio column, as an uneven distribution of data will affect the mean of the nearest k neighbours, and therefore the regression results. It may also be interesting to determine the count of community types; more larger communities lack a majority racial group compared to smaller communities (Farrell & Lee, 2016) while communities with more visitors tend to have higher crime rates (Dua & Graff, 2019) — this will be considered in the evaluation.

To begin, we apply the filter_all function to subset all rows with missing values, and obtain the number of rows.

In [7]:
cols <- colnames(crime_training)

missing_values <- crime_training |>
                    filter_all(any_vars(is.na(.))) |>
                    summarize(missing = n())
missing_values

missing
1 231

In order to calculate the mean and range, we filter out rows with missing values, and apply the map_dfr function to obtain a new dataframe with the calculated means, maximums, and minimums.

In [8]:
crime_data <- crime_training |>
                filter_all(all_vars(!is.na(.)))

crime_data

variable_mean <- crime_data |>
                    select(racepctblack:racePctHisp, crime_ratio) |>
                    map_dfr(mean, na.rm = TRUE) |>
                    rename(racepctblack__mean = racepctblack,
                           racePctWhite__mean = racePctWhite,
                           racePctAsian__mean = racePctAsian,
                           racePctHisp__mean = racePctHisp,
                           crime_ratio__mean = crime_ratio)

variable_mean

variable_max <- crime_data |>
                    select(racepctblack:racePctHisp, crime_ratio) |>
                    map_dfr(max, na.rm = TRUE) |>
                    rename(racepctblack__max = racepctblack,
                           racePctWhite__max = racePctWhite,
                           racePctAsian__max = racePctAsian,
                           racePctHisp__max = racePctHisp,
                           crime_ratio__max = crime_ratio)

variable_max

variable_min <- crime_data |>
                    select(racepctblack:racePctHisp, crime_ratio) |>
                    map_dfr(min, na.rm = TRUE) |>
                    rename(racepctblack__min = racepctblack,
                           racePctWhite__min = racePctWhite,
                           racePctAsian__min = racePctAsian,
                           racePctHisp__min = racePctHisp,
                           crime_ratio__min = crime_ratio)
variable_min



communityname           state racepctblack racePctWhite racePctAsian
1    BerkeleyHeightstownship NJ     1.37        91.78        6.50        
2    Tigardcity              OR     0.74        94.33        3.43        
3    Fargocity               ND     0.35        97.11        1.25        
4    Enfieldtown             CT     2.65        95.72        1.04        
5    Lebanoncity             NH     0.41        97.55        1.55        
6    Gorhamtown              ME     0.37        98.84        0.40        
7    Paragouldcity           AR     0.09        99.39        0.18        
8    Marioncity              OH     4.22        94.72        0.45        
9    OrmondBeachcity         FL     3.52        95.13        1.03        
10   Clintoncity             MS    17.14        81.71        0.92        
11   ChippewaFallscity       WI     0.11        99.03        0.34        
12   Newingtontown           CT     1.41        96.54        1.40        
13   Scotttownship           PA     0.59        97.87        1.37        
14   NewProvidenceborough    NJ     0.54        94.18        4.90        
15   GrandForkscity          ND     0.80        95.49        1.07        
16   NorthTonawandacity      NY     0.16        98.92        0.37        
17   Oshkoshcity             WI     0.79        96.25        2.18        
18   Princetonborough        NJ     8.51        82.51        7.07        
19   Menomoniecity           WI     1.20        94.56        3.50        
20   Alpharettacity          GA     2.55        95.22        1.36        
21   StevensPointcity        WI     0.51        95.73        2.79        
22   Woostercity             OH     3.49        95.03        1.17        
23   Eatontownborough        NJ    12.49        78.85        6.59        
24   Durangocity             CO     0.31        89.69        1.09        
25   Greendalevillage        WI     0.15        97.68        1.74        
26   LeagueCitycity          TX     5.13        88.12        2.32        
27   GloucesterCitycity      NJ     0.23        99.11        0.40        
28   Searcycity              AR     4.72        93.97        0.46        
29   Bellinghamcity          WA     0.79        93.76        2.78        
30   WestFargocity           ND     0.10        98.54        0.21        
⋮    ⋮                       ⋮     ⋮            ⋮            ⋮           
1400 Florencecity            SC    47.04        52.28         0.45       
1401 Chinocity               CA     9.75        67.31         3.44       
1402 Montgomerycity          AL    42.34        56.53         0.73       
1403 Montebellocity          CA     0.97        47.05        15.11       
1404 Shrewsburytown          MA     1.06        94.78         3.74       
1405 Clevelandcity           TN     7.17        91.55         0.47       
1406 Salinascity             CA     3.01        54.55         8.08       
1407 Vallejocity             CA    21.15        50.46        22.95       
1408 Lancastercity           CA     7.41        79.38         3.72       
1409 Sharontown              MA     3.18        94.41         2.22       
1410 Tularecity              CA     6.15        65.56         2.50       
1411 Phoenixcity             AZ     5.19        81.69         1.66       
1412 HuntingtonParkcity      CA     1.06        31.19         1.85       
1413 Magnoliacity            AR    34.08        65.15         0.43       
1414 SafetyHarborcity        FL     5.22        93.20         0.97       
1415 Woodwardcity            OK     0.20        95.12         0.49       
1416 LaQuintacity            CA     1.84        79.72         1.52       
1417 Walpoletown             MA     1.60        97.01         0.66       
1418 FortMyerscity           FL    32.16        64.24         0.81       
1419 Lawtoncity              OK    19.33        70.78         3.25       
1420 Pasadenacity            TX     0.97        83.73         1.57       
1421 FlowerMoundtown         TX     2.07        94.85         1.00       
1422 Wilm

racepctblack__mean racePctWhite__mean racePctAsian__mean racePctHisp__mean
1 9.1369             83.46732           2.894003           8.847936         
  crime_ratio__mean
1 0.1085395

racepctblack__max racePctWhite__max racePctAsian__max racePctHisp__max
1 89.95             99.53             57.46             95.29           
  crime_ratio__max
1 0.7016389

racepctblack__min racePctWhite__min racePctAsian__min racePctHisp__min
1 0                 7.21              0.06              0.12            
  crime_ratio__min
1 0.002784566

| Column Name         | Definition                                                      |
| :------------------:| :---------------------------------------------------------------|
| commmunityname      | the name of the community we obtained the crime data from
| state               | the state in which the community is located in 
| racepctblack        | the percentage of the population that identify as Black 
| racePctWhite        | the percentage of the population that identify as White 
| racePctAsian        | the percentage of the population that identify as Asian 
| racePctHisp         | the percentage of the population that identify as Hispanic 
| ViolentCrimesPerPop | the number of violent crimes per population 
| nonViolPerPop       | the number of non-violent crimes per population 
| crime_ratio         | the ratio of violent crime to non-violent crime per population 
| racepctblack__mean  | the mean of the percentage of the population that identify as Black
| racePctWhite__mean  | the mean of the percentage of the population that identify as White
| racePctAsian__mean  | the mean of the percentage of the population that identify as Asian
| racePctHisp__mean   | the mean of the percentage of the population that identify as Hispanic
| crime_ratio_mean    | the mean ratio of violent crime to non-violent crime per population
| racepctblack__max   | the maximum of the percentage of the population that identify as Black
| racePctWhite__max   | the maximum of the percentage of the population that identify as White
| racePctAsian__max   | the maximum of the percentage of the population that identify as Asian
| racePctHisp__max    | the maximum of the percentage of the population that identify as Hispanic
| crime_ratio_max     | the maximum ratio of violent crime to non-violent crime per population
| racepctblack__min   | the minimum of the percentage of the population that identify as Black
| racePctWhite__min   | the minimum of the percentage of the population that identify as White
| racePctAsian__min   | the minimum of the percentage of the population that identify as Asian
| racePctHisp__min    | the minimum of the percentage of the population that identify as Hispanic
| crime_ratio_min     | the minimum ratio of violent crime to non-violent crime per population

Upon inspecting the crime_data table, the community names contain one of: borough, town, township, and city. Hence, we can use the str_detect function to filter for rows matching the respective characters, and apply the summarize function to the result to obtain the count.

In [ ]:
borough_count <- crime_data |>
                filter(str_detect(communityname, "borough")) |>
                summarize(borough_count = n())

borough_count

town_count <- crime_data |>
                filter(str_detect(communityname, "town") & !str_detect(communityname, "township")) |>
                summarize(town_count = n())

town_count

township_count <- crime_data |>
                filter(str_detect(communityname, "township")) |>
                summarize(township_count = n())

township_count

city_count <- crime_data |>
                filter(str_detect(communityname, "city")) |>
                summarize(city_count = n())

city_count

In [ ]:
crime_summary_1 <- tibble(variable_mean, variable_max, variable_min) |>
                pivot_longer(cols = everything(),
                             names_to = "metric",
                             values_to = "value") |>
                separate(col = metric,
                         into = c("variable", "metric"),
                         sep = "__") |>
                pivot_wider(names_from = metric,
                            values_from = value)
                

crime_summary_1

crime_summary_2 <- tibble(city_count, township_count, town_count, borough_count, missing_values)

crime_summary_2

From the above tables, we note that only 231 rows have missing data, meaning that the quantity of data will be sufficient. Whites have the highest mean, while Asians have the lowest mean and smallest range. The range for African Americans, Hispanics and the crime ratio seem to be quite large compared to their relatively low means—visualizations could help determine whether the maximums and minimums are outliers. Note that the sum of the means for race is not 100%, meaning that some of the population may identify as more than one race. However, the difference is small, and can be ignored. Cities also seem to be the dominant community type, which may or may not affect the findings.

We now visualize the crime ratio against racial percentages for every race in scatter plots below. Note that from the summary tables above, the means for African American, Asian, and Hispanics are close to the minimums while the mean for Whites is close to the maximum. This means that a large proportion of the data will be clustered, however, we can avoid overplotting by applying a continous transformation to the x-axis for Whites and a logarithmic transformation for the other groups.

In [ ]:
options(repr.plot.width = 12, repr.plot.height = 10)
crime_plot_black <- crime_training |>
                    ggplot(aes(x = racepctblack, y = crime_ratio)) +
                    geom_point(alpha = 0.3) +
                    labs(x = "Population with African American heritage (%)", y = "Violent to non-violent crime ratio per 100K population", title = "Crime Ratio vs. African American Population") +
                    theme(text = element_text(size = 18)) +
                    scale_x_log10()

crime_plot_black

In [ ]:
crime_plot_white <- crime_training |>
                    ggplot(aes(x = racePctWhite, y = crime_ratio)) +
                    geom_point(alpha = 0.3) +
                    labs(x = "Population with Caucasian heritage (%)", y = "Violent to non-violent crime ratio per 100K population", title = "Crime Ratio vs. Caucasian Population") +
                    theme(text = element_text(size = 18)) +
                    scale_x_continuous()

crime_plot_white

In [ ]:
crime_plot_asian <- crime_training |>
                    ggplot(aes(x = racePctAsian, y = crime_ratio)) +
                    geom_point(alpha = 0.3) +
                    labs(x = "Population with Asian heritage (%)", y = "Violent to non-violent crime ratio per 100K population", title = "Crime Ratio vs. Asian Population") +
                    theme(text = element_text(size = 18)) +
                    scale_x_log10()

crime_plot_asian

In [ ]:
crime_plot_hispanic <- crime_training |>
                    ggplot(aes(x = racePctHisp, y = crime_ratio)) +
                    geom_point(alpha = 0.3) +
                    labs(x = "Population with Hispanic heritage (%)", y = "Violent to non-violent crime ratio per 100K population", title = "Crime Ratio vs. Hispanic Population") +
                    theme(text = element_text(size = 18)) +
                    scale_x_log10()

crime_plot_hispanic

The overall relationship seems to be weakly negative for Whites and weakly positive for the other groups, but we must consider the context behind our variables. Violent crime tends to be much less frequent than non-violent crime as seen from the above plots. Hence, the absolute differences between the predictions will be small, but a small increase in crime ratio can be considered significant in the real world.

EXPECTED OUTCOMES AND SIGNIFICANCE

Previous research has yielded a relationship between race and crime rate, where the white population is the most underrepresented among persons arrested for violent and non-violent crimes (Bonczar & Beck, 1997). Based on these findings, we can hypothesize that the ratio of violent to non-violent crime in predominantly white populations will be lower.

Through our findings, we hope to determine how much specific races are over or under represented for criminal involvement. This information is important as it links to the debate of whether or not there is overrepresentation in the justice and prison systems by certain races and the role of discrimination in such cases. If race can indeed predict the occurence of violent to non-violent crimes, it may only reinforce the dangers that stereotypes place on certain ethnic groups.

However, our question is merely predictive and not causal. We cannot say for certain whether race causes a higher incidence of violent crime compared to non-violent crime, or whether it is the only factor in a higher crime ratio. Further exploration could be conducted on determining if causal relationship exists, or perhaps on other demographics such as education or income and whether they can predict the crime ratio as well.

REFERENCES

Bonczar, T. P., & Beck, A. J. (1997, March 6). Lifetime likelihood of going to state or Federal Prison. Bureau of Justice Statistics. Retrieved October 28, 2022, from https://bjs.ojp.gov/library/publications/lifetime-likelihood-going-state-or-federal-prison 

Dua, D., & Graff, C. (2019). Communities and crime unnormalized data set. UCI Machine Learning Repository. Retrieved October 27, 2022, from https://archive.ics.uci.edu/ml/datasets/Communities+and+Crime+Unnormalized

Farrell, C. R., & Lee, B. A. (2016, December 26). No-majority communities: Racial diversity and change at the local level. Urban affairs review (Thousand Oaks, Calif.). Retrieved October 27, 2022, from https://pubmed.ncbi.nlm.nih.gov/31814691/

U.S. Census Bureau. U.S. Department of Commerce. (2022, October 13). Retrieved October 27, 2022, from https://www.commerce.gov/bureaus-and-offices/census

Weidner, R. R., & Schultz, J. (2019, August 13). Examining the relationship between U.S. incarceration rates and population health at the county level. SSM - population health. Retrieved October 27, 2022, from https://pubmed.ncbi.nlm.nih.gov/31485477/